packages/wallets/nitrogen/src/adapter.ts

```diff
@@ -0,0 +1,235 @@
+/* eslint-disable no-underscore-dangle */
+import type {
+    EventEmitter,
+    WalletName
+} from '@solana/wallet-adapter-base';
+import {
+    BaseMessageSignerWalletAdapter,
+    WalletAccountError,
+    WalletConnectionError,
+    WalletError,
+    WalletNotConnectedError,
+    WalletNotReadyError,
+    WalletPublicKeyError,
+    WalletReadyState,
+    WalletSignMessageError,
+    WalletSignTransactionError,
+    scopePollingDetectionStrategy,
+} from '@solana/wallet-adapter-base';
+import type { SendOptions, Transaction, TransactionSignature } from '@solana/web3.js';
+import { PublicKey } from '@solana/web3.js';
+import bs58 from 'bs58';
+
+interface NitrogenWalletEvents {
+    connect(...args: unknown[]): unknown;
+    disconnect(...args: unknown[]): unknown;
+}
+
+interface NitrogenWallet extends EventEmitter<NitrogenWalletEvents> {
+    _handleDisconnect(...args: unknown[]): unknown;
+    connect(): Promise<void>;
+    disconnect(): Promise<void>;
+    isConnected: boolean;
+    signAllTransactions(transactions: Transaction[]): Promise<Transaction[]>;
+    signAndSendTransaction(
+        transaction: Transaction,
+        options?: SendOptions
+    ): Promise<{ signature: TransactionSignature }>;
+    signMessage(message: Uint8Array): Promise<{ signature: Uint8Array }>;
+    signTransaction(transaction: Transaction): Promise<Transaction>;
+    isNitrogen?: boolean;
```

Comment:
seems redundant, doesn't seem to be used in code

Reply:
It's like https://github.com/solana-labs/wallet-adapter/blob/master/packages/wallets/phantom/src/adapter.ts#L27